x/auth/vesting/types/vesting_account.go

```diff
@@ -523,6 +523,72 @@ func (dva DelayedVestingAccount) String() string {
 	return out.(string)
 }
 
+//-----------------------------------------------------------------------------
+// Permanent Locked Vesting Account
+
+var _ vestexported.VestingAccount = (*PermanentLockedVestingAccount)(nil)
+var _ authtypes.GenesisAccount = (*PermanentLockedVestingAccount)(nil)
+
+// NewPermanentLockedVestingAccount returns a PermanentLockedVestingAccount
+func NewPermanentLockedVestingAccount(baseAcc *authtypes.BaseAccount, coins sdk.Coins) *PermanentLockedVestingAccount {
+	baseVestingAcc := &BaseVestingAccount{
+		BaseAccount:     baseAcc,
+		OriginalVesting: coins,
+		EndTime:         -1, // ensure EndTime is set to -1, as PermanentLockedVestingAccount's do not have an EndTime
+	}
+
+	return &PermanentLockedVestingAccount{baseVestingAcc}
+}
+
+// GetVestedCoins returns the total amount of vested coins for a permanent locked vesting
+// account. All coins are only vested once the schedule has elapsed.
+func (plva PermanentLockedVestingAccount) GetVestedCoins(_ time.Time) sdk.Coins {
+	return nil
+}
+
+// GetVestingCoins returns the total number of vesting coins for a permanent locked
+// vesting account.
+func (plva PermanentLockedVestingAccount) GetVestingCoins(_ time.Time) sdk.Coins {
+	return plva.OriginalVesting
+}
+
+// LockedCoins returns the set of coins that are not spendable (i.e. locked).
+func (plva PermanentLockedVestingAccount) LockedCoins(_ time.Time) sdk.Coins {
+	return plva.BaseVestingAccount.LockedCoinsFromVesting(plva.OriginalVesting)
```

Comment:
Why not just `return plva.OriginalVesting`?

Reply:
My understanding is LockedCoins = OriginalVesting - DelegatedCoins, i.e. the coins that are unused.

Reply:
Okay right